doc/policy/packages.md

```diff
@@ -72,3 +72,48 @@ test accepts):
      a competing package or transaction with a mutated witness, even though the two
      same-txid-different-witness transactions are conflicting and cannot replace each other, the
      honest package should still be considered for acceptance.
+
+### Package Fees and Feerate
+
+*Package Feerate* is the total modified fees (base fees + any fee delta from
+`prioritisetransaction`) divided by the total virtual size of all transactions in the package.
+If any transactions in the package are already in the mempool, they are not submitted again
+("deduplicated"), and thus excluded from this calculation.
+
+To meet the two feerate requirements of a mempool, i.e., the pre-configured minimum relay feerate
+(`minRelayTxFee`) and dynamic mempool minimum feerate, the total package feerate is used instead of
+the individual feerate. The individual transactions are allowed to be below feerate requirements if
+the package meets the feerate requirements. For example, the parent(s) in the package can have 0
+fees but be paid for by the child.
+
+*Rationale*: This can be thought of as "CPFP within a package," solving the issue of a parent not
+meeting minimum fees on its own. This allows L2 applications to adjust their fees at broadcast time
+instead of overshooting or risking getting stuck/pinned.
+
+*Rationale*: It would be incorrect to use the fees of transactions that are already in the mempool, as
+we do not want a transaction's fees to be double-counted.
+
+Implementation Note: Transactions within a package are always validated individually first, and
+package validation is used for the transactions that failed. Since package feerate is only
```

Comment:
Your description is correct at a high level, and is what I intend to communicate to the reader. To be nitpicky, we'll add transactions to the package for *any* policy reason, not just feerate, since `TxValidationResult` doesn't specify whether it's a fee-related or other policy issue.

> I think this algorithm generalizes to an arbitrary package DAG but I'm not completely sure.

While I wish this were true, there are cases where it doesn't work - see [example Q2](https://gist.github.com/glozow/dc4e9d5c5b14ade7cdfac40f43adb18a#always-try-individual-submission-first) in the package mempool accept gist. This is why the package-child-with-unconfirmed-parents restriction is so important.